packages/py-api/py_api/controllers/hackathon_teams_controller.py

```diff
@@ -0,0 +1,211 @@
+import json
+import random
+import string
+from typing import Any, Dict, List, Optional
+
+from bson.json_util import dumps
+from bson.objectid import ObjectId
+from fastapi.responses import JSONResponse
+from py_api.database.initialize import t_col
+from py_api.models.hackathon_teams_models import (
+    HackathonTeam,
+    MoveTeamMembers,
+    TeamType,
+    UpdateTeam,
+)
+from py_api.utilities.parsers import filter_none_values
+
+
+class TeamsController:
+    @classmethod
+    def create_team(cls, team_name: str, user_id: str = "", members_list: Optional[List[str]] = None) -> (
+            Dict[str, Any] | None
+    ):
+
+        team_type = TeamType.NORMAL
+
+        if not team_name:
+            # If no team_name is provided during registration, the participant is registering individually.
+            # We create a team of type RANDOM which should be filled with such participants
+
+            team_name = ''.join(
```

Comment:
How do we ensure that this randomly generated string doesn't match another team's name? I would suggest that you just count the number of teams of type `random` and just append + 1 to the the team's name. 

e.g. `RandomTeam 1` `RandomTeam 2` ...